clang/test/CodeGen/aie/peel-itercount.c

```diff
@@ -0,0 +1,35 @@
+// --- peel-itercount.c ---------------------------------------------------///
+// NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+// --------------------------------------------------------------------------///
+
+// RUN: %clang --target=aie2 -O2 -mllvm --unroll-force-peel-count=2 -S -emit-llvm %s -o - | FileCheck %s
```

Comment:
Would be good to have some generic tests (ideally `opt`) in llvm itself, that don't depend on AIE.
Maybe in llvm/test/Transforms/LoopUnroll?